### DISCUSSION TITLE
Channel attempting to execute code from another channel

### DISCUSSION THREAD

**Discussion by phatneff** [nextgenhealthcare/connect#6238] [General]
Hello, last night we ran into an issue when code from one channel was attempting to use code from a different channel.  Has anyone encountered this issue before? We are using version 4.4.2. See the attached readMe.txt file for more information.  
[readMe.txt](https://github.com/user-attachments/files/16055062/readMe.txt)

**Comment by pacmano1** [comment]
readme is empty.

**Comment by phatneff** [comment]
The two channel files and the server config file are attached here.
[360_CardioImaging_ORU_Outbound.txt](https://github.com/user-attachments/files/16055246/360_CardioImaging_ORU_Outbound.txt)
[ForEnhancement_Rad_ORU_Integration.txt](https://github.com/user-attachments/files/16055247/ForEnhancement_Rad_ORU_Integration.txt)
[serverConfig.txt](https://github.com/user-attachments/files/16055248/serverConfig.txt)

**Comment by phatneff** [comment]
How about now?

[readMe.txt](https://github.com/user-attachments/files/16055259/readMe.txt)

**Reply by pacmano1** [reply]
Yeah, but that serverconfig.txt won't import for me after file rename to .xml.

**Reply by phatneff** [reply]
What error are you getting when trying to import it?

**Reply by pacmano1** [reply]
Invalid server config.

**Reply by phatneff** [reply]
Try this one maybe.
[2024-07-01 Mirth Backup.txt](https://github.com/user-attachments/files/16056500/2024-07-01.Mirth.Backup.txt)

**Reply by pacmano1** [reply]
same problem (restoring to 4.5 though which should not matter).

**Reply by phatneff** [reply]
can you try changing all instances of "4.4.2" in the file to "4.5" and see if that makes a difference? I know it shouldn't in a channel file but not sure about the server config file.

**Reply by pacmano1** [reply]
really just wanted to see the code templates... they were not with the channels.

**Reply by phatneff** [reply]
[core.txt](https://github.com/user-attachments/files/16056649/core.txt)
here is the library with the code templates.

**Reply by pacmano1** [reply]
imported fine, thx

**Reply by pacmano1** [reply]
That's a lot of code to digest.  I wish I could help more that it's too much code for me to jump into.  To explicitly answer the question you asked though  - no, I have never seen code cross channels like that.  But then I haven't written the level of native calls to java you have in your code. I don't ever restart mirth other than for upgrades of the OS, DB or Mirth itself - not that it is relevant for your problem of course.

**Comment by jonbartels** [comment]
I read the README, you should EDIT your original post to include the text that is in README.txt so users do not have to download a file to read about the problem.

Check out this post from Odo - https://github.com/nextgenhealthcare/connect/discussions/4847#discussioncomment-1664403 . Can you try to run his snippet to see the assembled script that your failing channel is using?

**Reply by phatneff** [reply]
This was only a one time thing.  We can't replicate it ourselves.  We're just wondering if anyone has seen this before, knows why this may have occurred, and knows of a way to prevent it so it doesn't happen again.

**Reply by tonygermano** [reply]
Did you share what made you think that one channel was trying to run code from a different channel?

**Reply by phatneff** [reply]
> Did you share what made you think that one channel was trying to run code from a different channel?

It's in the readMe.txt file attached above.
